src/ImageSharp/Processing/PositionAwarePixelRowOperation.cs

```diff
@@ -7,10 +7,10 @@
 namespace SixLabors.ImageSharp.Processing
 {
     /// <summary>
-    /// A <see langword="delegate"/> representing a user defined pixel shader.
+    /// A <see langword="delegate"/> representing a user defined, position aware, processing delegate to use to modify image rows.
     /// </summary>
     /// <param name="span">The target row of <see cref="Vector4"/> pixels to process.</param>
     /// <param name="offset">The initial horizontal and vertical offset for the input pixels to process.</param>
     /// <remarks>The <see cref="Vector4.X"/>, <see cref="Vector4.Y"/>, <see cref="Vector4.Z"/>, and <see cref="Vector4.W"/> fields map the RGBA channels respectively.</remarks>
-    public delegate void PositionAwarePixelShader(Span<Vector4> span, Point offset);
+    public delegate void PositionAwarePixelRowOperation(Span<Vector4> span, Point offset);
```

Comment:
I don't think we need the verbosity here, the following might be enough:
```C#
public delegate void PixelRowOperation(Span<Vector4> span, Point offset);
```

We need to make sure that we will be able to extend the API with the following variant, since it is easier to use for non-bounded processing:
```C#
public delegate void PositionAwarePixelRowOperation(Span<Vector4> span, int row);
```

Reply:
How would the extension methods know which delegate is passed?

Reply:
I think C# compiler should be able to distinguish them based on signature, but it's worth a check.

Reply:
Figured out a nice way using generics. It also frees us up for using `TPixel` later.